rest_framework_json_api/parsers.py

```diff
@@ -139,6 +139,16 @@ def parse(self, stream, media_type=None, parser_context=None):
         if not data.get('id') and request.method in ('PATCH', 'PUT'):
             raise ParseError("The resource identifier object must contain an 'id' member")
 
+        if request.method in ('PATCH', 'PUT'):
+            kwarg_name = view.lookup_url_kwarg if view.lookup_url_kwarg else view.lookup_field
```

Comment:
```suggestion
            lookup_url_kwarg = view.lookup_url_kwarg or view.lookup_field
```

I would prefer to use this simpler form as it is also used in [DRF](https://github.com/encode/django-rest-framework/blob/0407a0df8a16fdac94bbd08d49143a74a88001cd/rest_framework/generics.py#L86) and also use same naming of variables (in this case `lookup_url_kwarg`)